### DISCUSSION TITLE
DomPDF & 4 columns

### DISCUSSION THREAD

**Discussion by olvakulin** [dompdf/dompdf#2415] [Q&A]
I have a large continuous text. In dompdf, it is normally formatted on several pages, but in one column. I need to formatted in 4 columns. column-count: 4 doesn't work. But the html file of course normally has 4 columns. Any help and ideas - thank you very much !!!

**Comment by bsweeney** [comment]
Do you have a sample of the HTML + CSS you're trying?

**Reply by olvakulin** [reply]
```
<style>
  @page {
            margin: 100px 20px 100px 20px ;
        }
        body {
            margin: 100px 20px 100px 20px;
            -webkit-column-count: 4;
            column-count: 4;
            font-size: 100%;
        }
        html {
            margin: 0px;
        }
</style>

<body>
<div>
 <?php 
foreach($songs as $song)
                      <div class="" style="background-color: tomato;color: white;">
                    {{ substr($song->artist, 0, 1) }}
                </div>
            
            <div class="">
            {{ $song->title }}<br/>
        </div>

?>
</div>
</body>
```

In html, the simplest cycle foreach is about 500 values. I cannot display it normally in the code. All browsers have 4 columns. In the pdf file - one

**Reply by bsweeney** [reply]
Ah, I see, my bad you absolutely indicated that this was for the `column-count` CSS property. I am unhappy to report that this is not yet supported by Dompdf. And, surprisingly, a request to support this column-count has not come up before.

**Reply by olvakulin** [reply]
I figured it out from different posts on different forums. But placing 4 columns in a table and calculating how many rows and then moving to a new column, etc. is unreasonably long, and I'm sure it won't work as it should. Plus I need each sheet in 4 columns plus the continuation of the text on the next sheet. Very sorry.

**Reply by bsweeney** [reply]
Yes at this time that particular layout is going to be a challenge to accomplish with Dompdf. If you can install something like wkhtmltopdf you'll get fuller support of the CSS spec. We hope to be there some day but progress is slow, particularly around implementation of new layout functionality.
